test/lib/middleware.test.js

```diff
@@ -0,0 +1,224 @@
+const chai = require('chai')
+const spies = require('chai-spies')
+const proxyquire = require('proxyquire')
+const express = require('express')
+const request = require('supertest')
+
+const expect = chai.expect
+chai.use(spies)
+
+const mockLogger = {
+  log: () => { },
+  debug: () => mockLogger.log(),
+  info: () => mockLogger.log(),
+  warn: () => mockLogger.log(),
+  error: () => mockLogger.log(),
+  fatal: () => mockLogger.log()
+}
+
+let MockLogger = () => mockLogger
+// Logger must be a constructor
+function Logger () {
+  return MockLogger.apply(this, arguments)
+}
+
+const { middleware } = proxyquire('../../lib/middleware', {
+  './logger': {
+    Logger
+  }
+})
+
+describe('middleware', () => {
+  const message = 'Some message'
+  const sandbox = chai.spy.sandbox()
+  let app
+
+  beforeEach(() => {
+    sandbox.on(mockLogger, ['log', 'debug', 'info', 'warn', 'error', 'fatal'])
+    app = express()
+    app.use(middleware)
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  describe('endpoint', () => {
+    it('only logs a log request once', done => {
+      request(app)
+        .post('/log/debug')
+        .send({ message })
+        .expect(201, () => {
+          expect(mockLogger.log).to.have.been.called.exactly(1)
+          done()
+        })
+    })
+
+    it('logs with the correct level', done => {
+      request(app)
+        .post('/log/warn')
+        .send({ message })
+        .expect(201, () => {
+          expect(mockLogger.debug).to.not.have.been.called()
+          expect(mockLogger.info).to.not.have.been.called()
+          expect(mockLogger.warn).to.have.been.called()
+          expect(mockLogger.error).to.not.have.been.called()
+          expect(mockLogger.fatal).to.not.have.been.called()
+          done()
+        })
+    })
+
+    it('logs with info if the level does not exist', done => {
+      request(app)
+        .post('/log/some_level')
+        .send({ message })
+        .expect(201, () => {
+          expect(mockLogger.debug).to.not.have.been.called()
+          expect(mockLogger.info).to.have.been.called()
+          expect(mockLogger.warn).to.not.have.been.called()
+          expect(mockLogger.error).to.not.have.been.called()
+          expect(mockLogger.fatal).to.not.have.been.called()
+          done()
+        })
+    })
+
+    it('responds with 500 if the log throws an error', done => {
+      const originalInfo = mockLogger.info
+      mockLogger.info = () => { throw new Error('some error') }
+      request(app)
+        .post('/log/info')
+        .send({ message })
+        .expect(500, () => {
+          mockLogger.info = originalInfo
+          done()
+        })
+    })
+  })
+
+  describe('middleware', () => {
```

Comment:
redundant describe

Reply:
Why? It even has a beforeEach assigned to it. It's to encapsulate all of the middleware specific tests, as compared to the endpoint (`POST /log/:level`) tests.

Reply:
A describe block is shouldn't be created to fulfill a `beforeEach` certain tests.
A better approch would be putting this before each  with the before each of all tests, 
that way when one wants to add a new test he have an dummy app which is functionality is full
and can be found at one single place at the beginning of the main describe